extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java

```diff
@@ -141,6 +141,20 @@ public void pause() {
         }
     }
 
+    @Override
+    public void pause(String identity) {
+        if (identity == null) {
+            LOGGER.warn("Cannot pause - identity is null");
+            return;
+        }
+        for (ScheduledTask task : scheduledTasks) {
+            if (identity.equals(task.trigger.id)) {
+                task.trigger.running = false;
+                return;
+            }
+        }
+    }
```

Comment:
Wondering whether using a Map would have been faster

Reply:
It would be faster for a large number of scheduled methods but I don't think it's worth the effort, i.e. the difference would be negligible in most cases.

Reply:
Okay. Let's keep this as is.